pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/ASTAnnotation.java

```diff
@@ -27,37 +26,58 @@ public ASTAnnotation(JavaParser p, int id) {
+    }
+
+    // @formatter:off
+    /**
+     * Returns true if this annotation suppresses the given rule.
+     * The suppression annotation is {@link SuppressWarnings}.
+     * This method returns true if this annotation is a SuppressWarnings,
+     * and if the set of suppressed warnings ({@link SuppressWarnings#value()})
+     * contains at least one of those:
+     * <ul>
+     *     <li>"PMD" (suppresses all rules);
+     *     <li>"PMD.rulename", where rulename is the name of the given rule;
+     *     <li>"all" (conventional value to suppress all warnings).
+     * </ul>
+     *
+     * <p>Additionnally, the following values suppress a specific set of rules:
+     * <ul>
+     *     <li>{@code "unused"}: suppresses rules like UnusedLocalVariable or UnusedPrivateField;
+     *     <li>{@code "serial"}: suppresses BeanMembersShouldSerialize and MissingSerialVersionUID;
+     * </ul>
+     *
+     * @param rule The rule for which to check for suppression
+     *
+     * @return True if this annotation suppresses the given rule
+     */
+    // @formatter:on
     public boolean suppresses(Rule rule) {
-        final String ruleAnno = "\"PMD." + rule.getName() + "\"";
-
-        if (jjtGetChild(0) instanceof ASTSingleMemberAnnotation) {
-            ASTSingleMemberAnnotation n = (ASTSingleMemberAnnotation) jjtGetChild(0);
-            return checkAnnototation(n, ruleAnno, rule);
-        } else if (jjtGetChild(0) instanceof ASTNormalAnnotation) {
-            ASTNormalAnnotation n = (ASTNormalAnnotation) jjtGetChild(0);
-            return checkAnnototation(n, ruleAnno, rule);
+
+        if (jjtGetChild(0) instanceof ASTMarkerAnnotation) {
+            return false;
         }
-        return false;
-    }
 
-    private boolean checkAnnototation(Node n, String ruleAnno, Rule rule) {
-        if (n.jjtGetChild(0) instanceof ASTName) {
-            ASTName annName = (ASTName) n.jjtGetChild(0);
-
-            if ("SuppressWarnings".equals(annName.getImage())
-                    || "java.lang.SuppressWarnings".equals(annName.getImage())) {
-                List<ASTLiteral> nodes = n.findDescendantsOfType(ASTLiteral.class);
-                for (ASTLiteral element : nodes) {
-                    if (element.hasImageEqualTo("\"PMD\"") || element.hasImageEqualTo(ruleAnno)
-                            // Check for standard annotations values
-                            || element.hasImageEqualTo("\"all\"")
-                            || element.hasImageEqualTo("\"serial\"") && serialRules.contains(rule.getName())
-                            || element.hasImageEqualTo("\"unused\"") && unusedRules.contains(rule.getName())) {
-                        return true;
-                    }
+        // if (SuppressWarnings.class.equals(getType())) { // typeres is not always on
+        if ("SuppressWarnings".equals(getAnnotationName()) || "java.lang.SuppressWarnings".equals(getAnnotationName())) {
```

Comment:
We could still use `SuppressWarnings.class.getSimpleName().equals(getAnnotationName())` instead of the Strings ... but for readability improvement, this probably should go into a simple private method `isSupressWarnings()`.